pytorch_lightning/trainer/training_loop.py

```diff
@@ -344,8 +353,15 @@ def _process_training_step_output(self, training_step_output, split_batch):
         # -----------------------------------------
         # no need for these checks in 1.0.0
         # TODO: remove checks in 1.0.0
```

Comment:
Unrelated to this PR but we should address this...

Reply:
Yes.